src/main/java/seedu/address/model/module/Grade.java

```diff
@@ -90,7 +90,7 @@ public Grade() {
      * Creates a new {@code Grade} object with value grade and State COMPLETE
      */
     public Grade(String grade) {
-        this(grade.toUpperCase(), State.COMPLETE);
+        this(grade.toUpperCase(), (EMPTY_VALUE.equals(grade)) ? State.INCOMPLETE : State.COMPLETE);
```

Comment:
Might want to look into problems with Yoda Condition. [Yoda Condition](http://knowthecode.io/yoda-conditions-yoda-not-yoda)
```suggestion
        this(grade.toUpperCase(), (grade.equals(EMPTY_VALUE)) ? State.INCOMPLETE : State.COMPLETE);
```